### PR TITLE
Run Azure DevOps builds against pull requests in master

### PR DESCRIPTION
**Pull request by echesakovMSFT**
This would provide default branches for PR and CI triggers. 

As soon as this merged (and we ready to enable AzDO-as-CI), we can uncheck boxes "Override YAML Pull Request trigger" and "Override YAML Continuous Integration trigger" in the build definition and control the behavior only via YAML file. 

Opt-in to use Mission Control for visualizing tests results and fail a corresponding step (and the whole job) if there any failed tests 

Reduce a load on arm32/arm64 machines be disabling the corresponding jobs' default triggers.

Keep other legs enabled.

### CODE REVIEW

**Comment by echesakovMSFT** [comment]
@dotnet-bot test ci

**Comment by echesakovMSFT** [comment]
@dotnet-bot test ci

**Comment by echesakovMSFT** [comment]
@dotnet-bot test ci

**Comment by echesakovMSFT** [comment]
@BruceForstall Is this a right way to disable default triggers for the Jenkins arm jobs?

File: netci.groovy [hunk 1]
```diff
@@ -1927,7 +1927,7 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
                         case 'innerloop':
                         case 'no_tiered_compilation_innerloop':
                             if (configuration == 'Checked') {
-                                isDefaultTrigger = true
+                                isDefaultTrigger = false
```
Comment (BruceForstall) [inline netci.groovy]: `isDefaultTrigger = false` is the default, so just remove the conditions.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -23,6 +23,11 @@ resources:
   - container: centos6_x64_build_image
     image: microsoft/dotnet-buildtools-prereqs:centos-6-376e1a3-20174311014331
 
+trigger: none
```
Comment (sbomer) [inline azure-pipelines.yml]: Won't this disable ci triggers? (Is that the intention)?

Comment (echesakovMSFT) [reply azure-pipelines.yml]: Yes, let's do transition in several steps - first PR validation, then CI

File: netci.groovy (hunk 1 above)
Comment (echesakovMSFT) [reply netci.groovy]: I left the innerloop jobs on purpose - if we find an issue with AzDev we would have a way to trigger the arm innerloop jobs. I will remove them as soon as the AzDev pipeline goes through enough testing

**Comment by echesakovMSFT** [comment]
@dotnet-bot test ci

File: netci.groovy (hunk 1 above)
Comment (echesakovMSFT) [reply netci.groovy]: > Is this change intended to move arm32/arm64 default triggered PR jobs from Jenkins to AzDev?

Not only, we are also enabling other jobs in AzDev against PRs but keeping them in Jenkins to monitor consistency between the corresponding legs

**Comment by echesakovMSFT** [comment]
The changes won't affect the CI legs, so merging without waiting for them to finish

**Pull request merged**